qemu.go

```diff
@@ -327,27 +327,28 @@ func networkModelToQemuType(model NetInterworkingModel) ciaoQemu.NetDeviceType {
 	}
 }
 
-func (q *qemu) appendNetworks(devices []ciaoQemu.Device, endpoints []Endpoint) []ciaoQemu.Device {
-	for idx, endpoint := range endpoints {
-		virtualEndpoint, ok := endpoint.(*VirtualEndpoint)
-		if ok {
-			devices = append(devices,
-				ciaoQemu.NetDevice{
-					Type:          networkModelToQemuType(virtualEndpoint.NetPair.NetInterworkingModel),
-					Driver:        ciaoQemu.VirtioNetPCI,
-					ID:            fmt.Sprintf("network-%d", idx),
-					IFName:        virtualEndpoint.NetPair.TAPIface.Name,
-					MACAddress:    virtualEndpoint.NetPair.TAPIface.HardAddr,
-					DownScript:    "no",
-					Script:        "no",
-					VHost:         true,
-					DisableModern: q.nestedRun,
-					FDs:           virtualEndpoint.NetPair.VMFds,
-				},
-			)
-		}
+var networkIndex = 0
 
+func (q *qemu) appendNetwork(devices []ciaoQemu.Device, endpoint Endpoint) []ciaoQemu.Device {
+	virtualEndpoint, ok := endpoint.(*VirtualEndpoint)
+	if ok {
```

Comment:
Again here we should switch on the interface type for making the code more generic.